test/unittest/StatisticsParticipantListener/StatisticsParticipantListenerTests.cpp

```diff
@@ -362,87 +480,109 @@ TEST_F(statistics_participant_listener_tests, new_participant_discovered)
     eprosima::fastrtps::rtps::ParticipantDiscoveryInfo info(data);
 
     // Expectation: The Participant is added to the database. We do not care about the given ID
-    InsertEntityArgs insert_args([&](
-                std::shared_ptr<Entity> entity)
+    InsertParticipantArgs insert_args([&](
+                const std::string& name,
+                const Qos& qos,
+                const std::string& guid,
+                const EntityId& domain_id,
+                const StatusLevel& status,
+                const AppId& app_id,
+                const std::string& app_metadata)
             {
-                EXPECT_EQ(entity->kind, EntityKind::PARTICIPANT);
-                EXPECT_EQ(entity->name, participant_name_);
-                EXPECT_EQ(std::dynamic_pointer_cast<DomainParticipant>(entity)->guid, participant_guid_str_);
-                EXPECT_EQ(std::dynamic_pointer_cast<DomainParticipant>(entity)->qos,
-                participant_proxy_data_to_backend_qos(info.info));
+                EXPECT_EQ(name, participant_name_);
+                EXPECT_EQ(qos, participant_proxy_data_to_backend_qos(info.info));
+                EXPECT_EQ(guid, participant_guid_str_);
+                EXPECT_EQ(domain_id, EntityId(0));
+                EXPECT_EQ(status, StatusLevel::OK_STATUS);
+                EXPECT_EQ(app_id, AppId::UNKNOWN);
+                EXPECT_EQ(app_metadata, "");
 
                 return EntityId(10);
             });
 
-    // Expectation: The host is created and given ID 13
-    InsertEntityArgs insert_args_host([&](
-                std::shared_ptr<Entity> entity)
+    EXPECT_CALL(database, insert_new_participant(_, _, _, _, _, _, _)).Times(1)
+            .WillOnce(Invoke(&insert_args, &InsertParticipantArgs::insert));
+
+    // Expectation: The host is created and given ID 13, the user is created and given ID 14 and the process is created and given ID 15
+    ProcessPhysicalArgs process_physical_args([&](
+                const std::string& host_name,
+                const std::string& user_name,
+                const std::string& process_name,
+                const std::string& process_pid,
+                bool& should_link_process_participant,
+                const EntityId& participant_id,
+                std::map<std::string, EntityId>& physical_entities_ids)
             {
-                EXPECT_EQ(entity->kind, EntityKind::HOST);
-                EXPECT_EQ(entity->name, host_name_);
-                EXPECT_EQ(entity->alias, host_name_);
-
-                return EntityId(13);
-            });
+                EXPECT_EQ(host_name, host_name_);
+                EXPECT_EQ(user_name, user_name_);
+                EXPECT_EQ(process_name, process_name_);
+                EXPECT_EQ(process_pid, process_name_);
```

Comment:
Sometimes processes follow one of these two patterns: `[command]:[PID] or [PID]`, Both cases are managed and tested (e.g: database_queue_tests.push_participant_data_wrong_processname_format).
The second pattern is used in this test. For clarity, it will be modified.